packages/shared/src/types/rsbuild.ts

```diff
@@ -59,4 +59,10 @@ export type RsbuildInstance<P extends RsbuildProvider = RsbuildProvider> = {
   >['pluginAPI']['onAfterStartProdServer'];
   onDevCompileDone: Awaited<ReturnType<P>>['pluginAPI']['onDevCompileDone'];
   onExit: Awaited<ReturnType<P>>['pluginAPI']['onExit'];
+
+  /**
+   * This method is used to override the Rsbuild default html-plugin (html-rspack-plugin).
+   *
+   */
+  setHtmlPlugin: (plugin: typeof import('html-webpack-plugin')) => void;
```

Comment:
this is a private helper, we can just export this method from `@rsbuild/core/provider`, so Rsbuild users will not use this method.